### PR TITLE
docs(table): add missing imports and improve example using http

### PR DESCRIPTION
**Pull request by rafaelss95**
- Add missing `OnInit` import.
- Remove unused operator imports.
- Remove unnecessary type annotations.
- Create a simple interface called `GithubApi` to transform the request.

[DEMO](https://plnkr.co/edit/NvCAxbuM8RTe3mhgGtyh)

### CODE REVIEW

**Comment by rafaelss95** [comment]
@andrewseguin can you review this?

File: src/material-examples/table-http/table-http-example.ts [hunk 1]
```diff
@@ -63,67 +68,49 @@ export class ExampleHttpDao {
...
-
-          return this.readGithubResult(result);
-        });
-
-
+      .startWith(null)
+      .switchMap(() => {
+        this.isLoadingResults = true;
+        return this.exampleDatabase.getRepoIssues(
+          this.sort.active, this.sort.direction, this.paginator.pageIndex
```
Comment (andrewseguin) [inline src/material-examples/table-http/table-http-example.ts]: Bring up the `);`, doesn't fit the rest of the style in the repo

Comment (rafaelss95) [inline src/material-examples/table-http/table-http-example.ts]: Done :+1:

**Comment by lemang** [comment]
can you add example for filter?

File: src/material-examples/table-http/table-http-example.ts (hunk 1 above)
Comment (rafaelss95) [inline src/material-examples/table-http/table-http-example.ts]: @andrewseguin can you apply the labels (`merge-ready` and `lgmt`) here?

Comment (rafaelss95) [inline src/material-examples/table-http/table-http-example.ts]: @andrewseguin can you merge this?

Comment (andrewseguin) [inline src/material-examples/table-http/table-http-example.ts]: @rafaelss95 Needs a rebase

Comment (rafaelss95) [inline src/material-examples/table-http/table-http-example.ts]: @andrewseguin done.

**Pull request merged**